YoCode/ProcessRunner.cs

```diff
@@ -5,26 +5,36 @@
 
 namespace YoCode
 {
-    class ProcessRunner
+    public class ProcessRunner
     {
+        private int timeout = 15000; //15 seconds
+        public bool TimedOut { get; set; }
         ProcessInfo procinfo;
         public string Output { get; set; }
+        public string ErrorOutput { get; set; }
 
-        public ProcessRunner(string processName,string workingDir,string arguments){
-            procinfo = setupProcessInfo(processName, workingDir, arguments);
+        public ProcessRunner(string processName,string workingDir,string arguments)
+        {
+            procinfo = SetupProcessInfo(processName, workingDir, arguments);
         }
 
         public void ExecuteTheCheck()
         {
             var p = new Process();
             p.StartInfo = SetProcessStartInfo(procinfo);
             p.Start();
+
+            if(!p.WaitForExit(timeout))
```

Comment:
If `timeout` is a TimeSpan then you would write `timeout.TotalMilliseconds` here.